server.js

```diff
@@ -13,39 +13,92 @@ app.listen('3001', () => {
   console.log('Server is now running at http://localhost:3001');
 })
 
+
 app.get('/api/v1/notes', (request, response) => {
   response.status(200).json(app.locals.notes);
 });
 
+
 app.post('/api/v1/notes', (request, response) => {
-  if (!request.body.type && !request.body.name) {
-    return response.status(422).send({
-      success: 'false',
-      message: 'name & type is required'
-    });
-  } else if (!request.body.name) {
-    return response.status(422).send({
-      success: 'false',
-      message: 'name is required'
-    });
-  } else if (!request.body.type) {
-    return response.status(422).send({
-      success: 'false',
-      message: 'type is required'
-    });
-  } else if (request.body.type && request.body.name) {
-    const note = {
-      id: shortid.generate(),
-      name: request.body.name,
-      type: request.body.type
+  const { notes } = app.locals;
+  const { name, type } = request.body;
+
+  if(!name && !type) return send422(response, 'Name and Type is required')
+  if(!name) return send422(response, 'Name is required')
+  if(!type) return send422(response, 'Type is required')
+
+  const note = { id: shortid.generate(), name, type }
+
+  notes.push(note);
+  return response.status(201).json(note);
+});
+
+
+app.put('/api/v1/notes/:id', (request, response) => {
+  const { name, type } = request.body;
+  const { id } = request.params;
+  const { notes } = app.locals;
+
+  let noteFound;
+  let noteIndex;
+
+  notes.map((note, index) => {
+    if (note.id == id) {
+      noteFound = note;
+      noteIndex = index;
+    }
+  });
+
+  if(!noteFound) return send400(response, 'Note is not found')
+  if(!name) return send400(response, 'Name is required')
+  if(!type) return send400(response, 'Type is required')
+
+  const updatedNote = {
```

Comment:
Let's convert this to the actual syntax of our note object instead of the 'pets' template